src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -50,11 +54,39 @@ public class HttpUtil {
   static final int DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT = 5; // minutes
   private static CloseableHttpClient httpClient;
 
+  private static PoolingHttpClientConnectionManager connectionManager;
+
+  private static IdleConnectionMonitorThread idleConnectionMonitorThread;
+
+  /**
+   * This lock is to synchronize on idleConnectionMonitorThread to avoid setting starting a thread
+   * which was already started. (To avoid {@link IllegalThreadStateException})
+   */
+  private static ReentrantLock idleConnectionMonitorThreadLock = new ReentrantLock(true);
+
+  private static final int DEFAULT_CONNECTION_TIMEOUT_MINUTES = 1;
+  private static final int DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT_MINUTES = 5;
```

Comment:
we already have these two above? line 53-54

Reply:
Ouch, thanks for catching, let me fix it.